include/boost/url/impl/const_string.ipp

```diff
@@ -110,6 +110,8 @@ const_string::
 factory::
 operator()(string_view s) const
 {
+    if (s.empty())
+        s = "";
```

Comment:
wait, what is this?

Reply:
We need to check if `s.empty()` (but we don't need the `s = "";`) because `s.data()` might be `nullptr` for an empty string. I'll push the changes when you're done reviewing.